controllers/ibmpowervscluster_controller_test.go

```diff
@@ -294,24 +309,397 @@ func TestIBMPowerVSClusterReconciler_Reconcile(t *testing.T) {
 func TestIBMPowerVSClusterReconciler_reconcile(t *testing.T) {
 	testCases := []struct {
 		name                string
-		powervsClusterScope *scope.PowerVSClusterScope
+		powervsClusterScope func() *scope.PowerVSClusterScope
 		clusterStatus       bool
+		expectedResult      ctrl.Result
+		expectedError       error
+		conditions          capiv1beta1.Conditions
 	}{
 		{
 			name: "Should add finalizer and reconcile IBMPowerVSCluster",
-			powervsClusterScope: &scope.PowerVSClusterScope{
-				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				return &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+				}
 			},
-			clusterStatus: false,
 		},
 		{
 			name: "Should reconcile IBMPowerVSCluster status as Ready",
-			powervsClusterScope: &scope.PowerVSClusterScope{
-				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
-					ObjectMeta: metav1.ObjectMeta{
-						Finalizers: []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				return &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Finalizers: []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+						},
+					},
+				}
+			},
+			clusterStatus: true,
+		},
+		{
+			name: "When Power VS zone does not support PER",
+			powervsClusterScope: func() *scope.PowerVSClusterScope {
+				return &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Finalizers:  []string{infrav1beta2.IBMPowerVSClusterFinalizer},
+							Annotations: map[string]string{infrav1beta2.CreateInfrastructureAnnotation: "true"},
+						},
 					},
+				}
+			},
+			expectedError: errors.New("PowerVS zone is not set"),
```

Comment:
wondering if the expected error is inline with expectation of the test case (as per name)
`name: "When Power VS zone does not support PER",`

Reply:
Updated the test case.